### PR TITLE
Gem worth

### PR DESCRIPTION
**Pull request by benjy3gg**
calculate the worth of gems in stashes
- assume the level of gem is level 20 when level > 19
- assume the quality of gem is quality 20 when quality > 16

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -898,9 +927,30 @@ class ReportBuilder {
...
       }
 
+      let getLineItemQuality = (lineItem) => {
+        return lineItem.quality || 0
+      }
+
+      let rateExists = (type, item) => {
+        let litemName = getLineItemName(item).toLowerCase()
+        let gemLevel = item.gemLevel
+        let gemQuality = item.gemQuality
```
Comment (nijikokun) [inline src/components/App.js]: These two variables should be inside of the type check below

File: src/components/App.js [hunk 2]
```diff
@@ -898,9 +927,30 @@ class ReportBuilder {
...
+        if(type == "gem") {
```
Comment (nijikokun) [inline src/components/App.js]: Space between `if` and `(`

File: src/components/App.js [hunk 3]
```diff
@@ -898,9 +927,30 @@ class ReportBuilder {
...
+          return rates.find(value => {
+            value.lname === litemName && value.gemLevel == gemLevel && value.gemQuality == gemQuality
```
Comment (nijikokun) [inline src/components/App.js]: Missing `return` keyword

File: src/components/App.js [hunk 4]
```diff
@@ -937,6 +987,55 @@ class ReportBuilder {
...
+        quality = 0
+      }
+
+      // ninja only return level = 0, 20, 21 so we set level 19 to level 20 and all other we set to 0
+      // maybe make the check check for level 18 because it is nearly 20?
+      if(level === 19) {
+        level = 20
+      } else if (level < 19) {
+        level = 1
+      }
```
Comment (nijikokun) [inline src/components/App.js]: This feels dubious to me, I would leave out gems that aren't these values.

File: src/components/App.js [hunk 5]
```diff
@@ -937,6 +987,55 @@ class ReportBuilder {
...
+      return [level, quality, item];
+    }
+
+    // Helpers
+    let getGemObject = (item) => {
+      let gemName = item.typeLine.toLowerCase()
+      let gemQuality
+      let gemLevel
+      let gemCorrupted = item.corrupted || false;
+      [gemLevel, gemQuality, item] = getGemProperties(item)
```
Comment (nijikokun) [inline src/components/App.js]: Do the assign at the same time, not before 

`let [] = ...`

**Comment by benjy3gg** [comment]
should be fixed

**Pull request closed without merging**